### PR TITLE
[MXNET-307] Add .md tutorials to .ipynb for CI integration

### PR DESCRIPTION
**Pull request by ThomasDelteil**
## Description ##
- **+11** tutorials downloadable as jupyter notebooks and ready to integrate in the CI
- Added  where necessary to beautify the notebooks
- updated notebooks to remove `!bash`-like commands so they can run more easily on Windows
- updated paths to be cross-platform
- tested against master using the tutorial test runner (see log) on python 2 and python 3 (integration in CI coming up)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

```
Test Summary Start
31 tutorials tested:
basic/ndarray
basic/ndarray_indexing
basic/symbol
basic/module
basic/data
gluon/customop
gluon/data_augmentation
gluon/datasets
gluon/ndarray
gluon/mnist
gluon/autograd
gluon/gluon
gluon/hybrid
nlp/cnn
onnx/super_resolution
onnx/fine_tuning_gluon
onnx/inference_on_onnx_model
python/matrix_factorization
python/linear-regression
python/mnist
python/predict_image
python/data_augmentation
python/data_augmentation_with_masks
python/kvstore
python/types_of_data_augmentation
sparse/row_sparse
sparse/csr
sparse/train
speech_recognition/ctc
unsupervised_learning/gan
vision/large_scale_classification

0 tests failed:
Test Summary End
Stats start
[Passed: 31 of 31]
Total time: 514.86s
Stats end
```

### CODE REVIEW

**Comment by ThomasDelteil** [comment]
@thomelane if you have time for a review 😄

File: tests/nightly/test_tutorial_config.txt [hunk 1]
```diff
@@ -1,20 +1,31 @@
...
+python/linear-regression
+python/mnist
+python/predict_image
+python/data_augmentation
+python/data_augmentation_with_masks
+python/kvstore
+python/types_of_data_augmentation
 sparse/row_sparse
 sparse/csr
-sparse/train
```
Comment (rahul003) [inline tests/nightly/test_tutorial_config.txt]: Can we pick them automatically so we don't need to keep updating this file?

Comment (ThomasDelteil) [reply tests/nightly/test_tutorial_config.txt]: Indeed, going forward, there will be a one individual test per tutorial, to allow the use of annotation like `@highCpu`, `@hihMemory`. And there will be an integration test that will check that each notebook has been added to the test suite. 

This will be part of my next PR, as part of this work of integrating tutorials to the CI 😃

Comment (marcoabreu) [reply tests/nightly/test_tutorial_config.txt]: Yeah, sorry @rahul003. Thomas suggested the same as you, but we will have to be able to annotate the tests and give the possibility to add custom testing behaviour - e.g. methods to validate a tutorial which are different for each one.

File: docs/tutorials/gluon/datasets.md [hunk 2]
```diff
@@ -175,58 +180,51 @@ for epoch in range(epochs):
...
+    mx.test_utils.download("{}{}".format(data_url, archive_file), dirname = data_folder)
+    print('Extracting {} in {}...'.format(archive_file, data_folder))
+    tar = tarfile.open(archive_path, "r:gz")
+    tar.extractall(data_folder)
+    tar.close()
+    print('Data extracted.')
 ```
 
-After downloading and extracting the data archive, we seperate the data into training and test sets (50:50 split), and place images of the same class into the same folders, as required for using [`ImageFolderDataset`](https://mxnet.incubator.apache.org/api/python/gluon/data.html?highlight=imagefolderdataset#mxnet.gluon.data.vision.datasets.ImageFolderDataset).
+After downloading and extracting the data archive, we have two folders: `data/101_ObjectCategories` and `data/101_ObjectCategories_test`. We load the data into a training and testing dataset  [`ImageFolderDataset`](https://mxnet.incubator.apache.org/api/python/gluon/data.html?highlight=imagefolderdataset#mxnet.gluon.data.vision.datasets.ImageFolderDataset).
```
Comment (thomelane) [inline docs/tutorials/gluon/datasets.md]: We load the data into separate training and testing [`ImageFolderDataset`](https://mxnet.incubator.apache.org/api/python/gluon/data.html?highlight=imagefolderdataset#mxnet.gluon.data.vision.datasets.ImageFolderDataset)s.

File: docs/tutorials/gluon/datasets.md [hunk 3]
```diff
@@ -245,7 +243,7 @@ As with the Fashion MNIST dataset the labels will be integer encoded. You can us
 
 
 ```python
-sample_idx = 888
+sample_idx = 539
```
Comment (thomelane) [inline docs/tutorials/gluon/datasets.md]: Does the image need to be updated now?

File: docs/tutorials/python/data_augmentation.md [hunk 4]
```diff
@@ -49,13 +47,21 @@ One of the most convenient ways to augment your image data is via arguments of [
 We show a simple example of this below, after creating an `images.lst` file used by the [`ImageIter`](https://mxnet.incubator.apache.org/api/python/image/image.html?highlight=imageiter#mxnet.image.ImageIter). Use [`tools/im2rec.py`](https://github.com/apache/incubator-mxnet/blob/master/tools/im2rec.py) to create the `images.lst` if you don't already have this for your data.
 
 ```python
-!echo -e "0\t0.000000\timages/0.jpg" > ./data/images.lst
+path_to_image = os.path.join("images","0.jpg")
+index = 0
+label = "0.000000"
+list_file_content = "{}\t{}\t{}".format(index, label, path_to_image)
```
Comment (thomelane) [inline docs/tutorials/python/data_augmentation.md]: Could use {:5f}

File: docs/tutorials/python/types_of_data_augmentation.md [hunk 5]
```diff
@@ -335,7 +332,7 @@ And lastly, you can use [`mxnet.image.RandomOrderAug`](https://mxnet.incubator.a
 ```python
 example_image_copy = example_image.copy()
 aug_list = [
-    mx.image.RandomCropAug(size=(50, 50)),
+    mx.image.RandomCropAug(size=(250, 250)),
```
Comment (thomelane) [inline docs/tutorials/python/types_of_data_augmentation.md]: Why change? And do we need to update image?

File: docs/tutorials/speech_recognition/ctc.md [hunk 6]
```diff
@@ -1,5 +1,11 @@
 # Connectionist Temporal Classification
 
+```python
+
+import mxnet as mx
+print(mx.__version__)
```
Comment (thomelane) [inline docs/tutorials/speech_recognition/ctc.md]: Should remove as don't have this anywhere else, and it's result isn't shown either.

File: docs/tutorials/vision/large_scale_classification.md [hunk 7]
```diff
@@ -11,6 +11,11 @@ Training a neural network with a large number of images presents several challen
 $ pip install opencv-python
 ```
 
+```python
+import mxnet as mx
+print(mx.__version__)
```
Comment (thomelane) [inline docs/tutorials/vision/large_scale_classification.md]: Same as before.

File: docs/tutorials/gluon/datasets.md (hunk 3 above)
Comment (ThomasDelteil) [reply docs/tutorials/gluon/datasets.md]: no I found the same by doing a bisect

File: docs/tutorials/gluon/datasets.md (hunk 2 above)
Comment (ThomasDelteil) [reply docs/tutorials/gluon/datasets.md]: will update

File: docs/tutorials/python/types_of_data_augmentation.md (hunk 5 above)
Comment (ThomasDelteil) [reply docs/tutorials/python/types_of_data_augmentation.md]: the generated image was really random, this one always shows at least a part of the giraffe, which makes more sense IMO. The image can remain since it is a sensible one.

File: docs/tutorials/vision/large_scale_classification.md (hunk 7 above)
Comment (ThomasDelteil) [reply docs/tutorials/vision/large_scale_classification.md]: will update

File: docs/tutorials/python/data_augmentation.md (hunk 4 above)
Comment (ThomasDelteil) [reply docs/tutorials/python/data_augmentation.md]: I find it slightly less visual that way but will update
